jofing-forDevReview/ComponentDialogs/Startup.cs

```diff
@@ -0,0 +1,87 @@
+﻿namespace ComponentDialogs
+{
+    using System.Linq;
+    using Microsoft.AspNetCore.Builder;
+    using Microsoft.AspNetCore.Hosting;
+    using Microsoft.Bot.Builder;
+    using Microsoft.Bot.Builder.BotFramework;
+    using Microsoft.Bot.Builder.Dialogs;
+    using Microsoft.Bot.Builder.Integration;
+    using Microsoft.Bot.Builder.Integration.AspNet.Core;
+    using Microsoft.Bot.Builder.TraceExtensions;
+    using Microsoft.Extensions.Configuration;
+    using Microsoft.Extensions.DependencyInjection;
+    using Microsoft.Extensions.Options;
+
+    public class Startup
+    {
+        public Startup(IHostingEnvironment env)
+        {
+            var builder = new ConfigurationBuilder()
+                .SetBasePath(env.ContentRootPath)
+                .AddJsonFile("appsettings.json", optional: true, reloadOnChange: true)
+                .AddJsonFile($"appsettings.{env.EnvironmentName}.json", optional: true)
+                .AddEnvironmentVariables();
+
+            Configuration = builder.Build();
+        }
+
+        public IConfiguration Configuration { get; }
+
+        public void ConfigureServices(IServiceCollection services)
+        {
+            services.AddBot<HotelBot>(options =>
+            {
+                options.CredentialProvider = new ConfigurationCredentialProvider(Configuration);
+                options.OnTurnError = async (context, exception) =>
+                {
+                    await context.TraceActivityAsync("Bot Exception", exception);
+                    await context.SendActivityAsync("Sorry, it looks like something went wrong!");
+                };
+
+                // We're using both conversation and user state.
+                IStorage dataStore = new MemoryStorage();
+                var convState = new ConversationState(dataStore);
+                var userState = new UserState(dataStore);
+                options.Middleware.Add(new BotStateSet(convState, userState));
+            });
+
+            // Create and register the dialog state accessor off of conversation state.
+            services.AddSingleton(sp =>
+            {
+                var options = sp.GetRequiredService<IOptions<BotFrameworkOptions>>().Value;
+                var stateSet = options.Middleware.OfType<BotStateSet>().FirstOrDefault();
+                var convState = stateSet.BotStates.OfType<ConversationState>().FirstOrDefault();
```

Comment:
>stateSet [](start = 32, length = 8)

Not sure that i'd interrogate the BotStateSet middleware component to discover your State providers (in case people choose not to use BotStateSet).  

Instead use Options (which is new, and for this purpose):
   options.State.Add(convState);
   options.State.Add(userState);

Reply:
why do we even need to options.State at all? Why can't we just DI the state objects into the Bot object - wouldn't that just be simpler?

Reply:
@daveta , I missed this API change. I'll update my code to use `options.State`.

@johnataylor , I think the state middleware still needs to be registered on the adapter, thus adding it first to the options.